pipeline/lib/storage.ml

```diff
@@ -113,3 +113,29 @@ let record_success ~serial_id (db : Postgresql.connection) =
 
 let record_success ~serial_id ~conninfo =
   Db_util.with_db ~conninfo (record_success ~serial_id)
+
+let mark_closed_pull_requests ~open_pulls (db : Postgresql.connection) =
+  Logs.debug (fun log -> log "Updating open and closed pulls...");
+  let open_where_clause =
+    String.concat " OR "
+    @@ List.map
+         (fun (repo_id, pull_number) ->
+           Fmt.str {|(repo_id = '%s' AND pull_number = %d)|} repo_id pull_number)
+         open_pulls
+  in
+  let query =
+    Fmt.str
+      {|UPDATE benchmark_metadata
+        SET pull_closed = true
+        WHERE pull_number is NOT NULL;
+
+        UPDATE benchmark_metadata
+        SET pull_closed = false
+        WHERE %s;
+      |}
```

Comment:
Perhaps this could be a one-shot update (rather than two), with something like:
```sql
UPDATE ...
SET pull_closed = ( pull_number is not null AND NOT ( equal_to_one_of_the_open_pr ) )
```
But this actually removes the `NULL` state for branches... I personally have a hard time with negations, what do you think of replacing `pull_closed` with an `is_open` boolean not null? (such that the main branch can be `is_open=true`, and keeping the intuition that "true is good" that works well with filters etc)

Reply:
This makes sense too. Matching the flag with the data being fetched from the API would make things easier to understand. I'll change this!